### ISSUE TITLE
is.fn(async function () {}) returns false

### ISSUE THREAD

**Issue by chrisyip** [enricomarino/is#28]
```sh
$ node --harmony
> process.versions.node
'7.5.0'
> require('is').fn(async function () {})
false
> Object.prototype.toString.call(async function () {})
'[object AsyncFunction]'
> typeof async function () {}
'function'
```

The problem is `toStr.call(asyncFunction)` returns `[object AsyncFunction]` instead `[object Function]`.

Maybe it would be better to use `typeof fn === 'function'` instead `toStr.call(fn) === '[object Function]'`?

**Comment by ljharb** [comment]
There are browsers where `typeof` a regex returns "function", so that wouldn't work.

Since `--harmony` should *never* ever be used, and there isn't yet a node that has async functions unflagged, this isn't urgent just yet for AsyncFunction.

However, this bug does likely exist for generator functions too, so I'll address it ASAP.

**Comment by chrisyip** [comment]
Oh, I got it, browser compatibility. Thanks.

**Comment by chrisyip** [comment]
Async function is unflagged in node v7.6, maybe it's time to support detecting async function?

**Comment by ljharb** [comment]
@chrisyip that was fixed in f13acba 11 days ago, it just hasn't been released yet.

**Comment by ljharb** [comment]
v3.2.1 published.
